flatdata-py/flatdata/archive.py

```diff
@@ -109,7 +108,14 @@ def _is_archive():
 
     def _check_non_subarchive_schema(self, name, resource):
         if resource.container._is_archive():
-            return
-        actual_schema = self._resource_storage.get(name + _SCHEMA_EXT).read().decode()
+            return True
+
+        storage = self._resource_storage.get(name + _SCHEMA_EXT, resource.is_optional)
```

Comment:
I would be happier if this change is done in a separate method. Otherwise it turns method into a `_check_non_subarchive_schema_and_return_true_on_subarchives_or_optional`. A bit too convoluted to my taste.

Reply:
Please see my proposition in the following commit.